sql/src/main/java/io/crate/analyze/AnalyzedTableElements.java

```diff
@@ -142,6 +142,9 @@ private void expandColumn(AnalyzedColumnDefinition column) {
     }
 
     private static void addPrimaryKeys(List<String> primaryKeys, AnalyzedColumnDefinition column) {
+        if (column.isArrayOrInArray() && column.isPrimaryKey()) {
+            throw new IllegalArgumentException("Primary key constraint cannot be used within an array type");
```

Comment:
why not move the isArrayOrInArray check into the if block below which is checking for isPrimaryKey ?